src/contracts/middleware/BLSSignatureChecker.sol

```diff
@@ -1,508 +1,218 @@
 // SPDX-License-Identifier: BUSL-1.1
 pragma solidity =0.8.12;
 
-import "../interfaces/IBLSRegistry.sol";
-import "../libraries/BytesLib.sol";
+import "../interfaces/IBLSRegistryCoordinatorWithIndices.sol";
 import "../libraries/MiddlewareUtils.sol";
 import "../libraries/BN254.sol";
+import "../libraries/BitmapUtils.sol";
 
 /**
  * @title Used for checking BLS aggregate signatures from the operators of a `BLSRegistry`.
  * @author Layr Labs, Inc.
  * @notice Terms of Service: https://docs.eigenlayer.xyz/overview/terms-of-service
  * @notice This is the contract for checking the validity of aggregate operator signatures.
  */
-abstract contract BLSSignatureChecker {
+contract BLSSignatureChecker {
+    using BN254 for BN254.G1Point;    
+
```

Comment:
it looks like we have an interface defined for this contract, but the contract isn't inheriting from the interface. This means things like the struct definitions are duplicated in both -- we should define the structs in the interface and have this contract inherit from the interface.

Reply:
resolved https://github.com/Layr-Labs/eigenlayer-contracts/pull/35/commits/53db6100c35ef1e3b3b7f411ccf81749f45dd2f2